core/arch/arm/plat-vexpress/pm_a32.S

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2018, Linaro Limited
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <asm.S>
+#include <arm.h>
+#include <arm32_macros.S>
+#include <kernel/unwind.h>
+#include <platform_config.h>
+
+#if (CFG_TEE_CORE_NB_CORE > 1)
+/*
+ * Current Arm trusted firmware for the AArch32 paltforms traps the secondary
+ * boot cores in Monitor mode hence the early entry below must moves these
+ * cores into Supervisor mode.
+ */
+
+/* Must preserve r4, r5, r6 and r7, can clobber other registers, no stack */
+FUNC plat_cpu_reset_early , :
+UNWIND(	.fnstart)
+	mov	r8, lr
+	bl	__get_core_pos
+	cmp	r0, #0
+	beq	1f
+
+	cps	#CPSR_MODE_SVC
```

Comment:
Since we're always expecting to run in secure SVC mode, why not add this single instruction unconditionally before the call to `plat_cpu_reset_early()`. I don't think the `isb` is needed either.

Reply:
ok, i'll look at this option.